### DISCUSSION TITLE
Ошибка "PHP message: The WC_Customer::get_postcode function is deprecated since version 3.0. Replace with WC_Customer::get_billing_postcode"

### DISCUSSION THREAD

**Discussion by AleheyGo** [cdek-it/wordpress#29] [General]
Привет. Ковырял логи php, наткнулся на сабж:

[error] 984#984: *111582 FastCGI sent in stderr: "PHP message: The WC_Customer::get_postcode function is deprecated since version 3.0. Replace with WC_Customer::get_billing_postcode" while reading response header from upstream, client: xxx.xxx.xx.xxx, server: xx.xxx.xxx.xx, request: "POST /?wc-ajax=update_order_review HTTP/2.0", upstream: "fastcgi://unix:/var/www/php-fpm/2.sock:", host: "xxx.xxx", referrer: "https://xxx.xxx/checkout/".

Я не прогер, но на пару с GPT нашел проблему:
в cdekdelivery/src/UI/CheckoutMap.php на 24 строке заменил:
$postcodeInput = CheckoutHelper::getValueFromCurrentSession('postcode');  >>>>>> $postcodeInput = CheckoutHelper::getValueFromCurrentSession('billing_postcode');

После этого ошибки не наблюдаю.

**Comment by vermorag** [comment]
Добрый день. Это не ошибка, а предупреждение. Исправлению не подлежит, поскольку это необходимо для обратной совместимости со старыми версиями WP/WC. Плагин сначала проверяет `postcode`, потом `shipping_postcode`, потом `billing_postcode` поля.

**Reply by AleheyGo** [reply]
Отлично, на всякий случай написал. Спасибо за пояснение!
